crates/daphne-service-utils/src/durable_requests/bindings/aggregate_store_v2.rs

```diff
@@ -0,0 +1,195 @@
+        agg_job_id.encode_to_builder(builder.reborrow().init_agg_job_id());
+    }
+}
+
+impl CapnprotoPayloadDecode for PutRequest {
+    type Reader<'a> = aggregate_store_v2_capnp::put_request::Reader<'a>;
+
+    fn decode_from_reader(reader: Self::Reader<'_>) -> capnp::Result<Self> {
+        Ok(Self {
+            agg_share_delta: <_>::decode_from_reader(reader.get_agg_share_delta()?)?,
+            agg_job_id: <_>::decode_from_reader(reader.get_agg_job_id()?)?,
+        })
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use prio::{
+        codec::Decode,
+        field::{Field128, Field64, FieldElement, FieldPrio2},
+        vdaf::AggregateShare,
+    };
+    use prio_draft09::{
+        codec::Decode as DecodeDraft09,
+        field::{
+            Field128 as Field128Draft09, Field64 as Field64Draft09,
+            FieldElement as FieldElementDraft09, FieldPrio2 as FieldPrio2Draft09,
+        },
+        vdaf::AggregateShare as AggregateShareDraft09,
+    };
+    use rand::{thread_rng, Rng};
+
+    use crate::capnproto::{CapnprotoPayloadDecodeExt as _, CapnprotoPayloadEncodeExt as _};
+
+    use super::*;
+    use daphne::vdaf::VdafAggregateShare;
+
+    #[test]
+    fn serialization_deserialization_round_trip_draft09() {
+        let mut rng = thread_rng();
+        for len in 0..20 {
+            let test_data = [
+                VdafAggregateShare::Field64Draft09(AggregateShareDraft09::from(
+                    (0..len)
+                        .map(|_| {
+                            Field64Draft09::get_decoded(
+                                &rng.gen::<[_; Field64Draft09::ENCODED_SIZE]>(),
+                            )
+                            .unwrap()
+                        })
+                        .collect::<Vec<_>>(),
+                )),
+                VdafAggregateShare::Field128Draft09(AggregateShareDraft09::from(
+                    (0..len)
+                        .map(|_| {
+                            Field128Draft09::get_decoded(
+                                &rng.gen::<[_; Field128Draft09::ENCODED_SIZE]>(),
+                            )
+                            .unwrap()
+                        })
+                        .collect::<Vec<_>>(),
+                )),
+                VdafAggregateShare::Field32Draft09(AggregateShareDraft09::from(
+                    (0..len)
+                        .map(|_| {
+                            // idk how to consistently generate a valid FieldPrio2 value, so I just
+                            // retry until I hit a valid one. Doesn't usualy take too long.
+                            (0..)
+                                .find_map(|_| {
+                                    FieldPrio2Draft09::get_decoded(&rng.gen::<[_; 4]>()).ok()
+                                })
+                                .unwrap()
+                        })
+                        .collect::<Vec<_>>(),
```

Comment:
Likewise below, except you `prio::field::random_vector()` is infallible.
```suggestion
                VdafAggregateShare::Field64Draft09(AggregateShareDraft09::from(
                    random_vector(len).unwrap(),
                )),
                VdafAggregateShare::Field128Draft09(AggregateShareDraft09::from(
                    random_vector(len).unwrap(),
                )),
                VdafAggregateShare::Field32Draft09(AggregateShareDraft09::from(
                    random_vector(len).unwrap(),
                )),
```